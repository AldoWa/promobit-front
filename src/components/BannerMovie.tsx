import {
  Box,
  CircularProgress,
  CircularProgressLabel,
  Container,
  HStack,
  Heading,
  Image,
  Skeleton,
  Text,
  VStack,
} from '@chakra-ui/react';
import React from 'react';

export interface BannerMovieProps {
  title: string;
  imageUrl: string;
  avaliation: number;
  subtitle: string;
  overview: string;
  credits: Array<{
    type: string;
    name: string;
  }>;
  screenPlay: Array<{
    type: string;
    name: string;
  }>;
}

const BannerMovie: React.FC<BannerMovieProps> = ({
  title,
  imageUrl,
  credits,
  avaliation,
  subtitle,
  overview,
  screenPlay,
}) => {
  return (
    <Box height={600} bg="purple.900" data-testid="banner-movie-box">
      <Container maxW="1250px" data-testid="banner-movie-container">
        <HStack pt="4.5rem" alignItems="flex-start" gap={8}>
          <Image
            w={383}
            h={574}
            src={imageUrl}
            alt="Banner movie"
            fallback={<Skeleton w={383} h={574} borderRadius={4} />}
          ></Image>
          <VStack alignItems="start" gap={0}>
            <Heading
              as="h3"
              size="xl"
              color="white"
              data-testid="banner-movie-title"
            >
              {title}
            </Heading>
            <Text
              color="white"
              fontSize="1.125rem"
              mt={2}
              lineHeight={6}
              fontWeight={400}
              data-testid="banner-movie-subtitle"
            >
              {subtitle}
            </Text>
            <HStack mt={4}>
              <Box borderRadius="100%">
                <CircularProgress
                  size={59}
                  value={avaliation}
                  color="green.300"
                  trackColor="transparent"
                >
                  <CircularProgressLabel
                    color="green.300"
                    fontSize={16}
                    fontWeight={700}
                    data-testid="banner-movie-avaliation"
                  >
                    {avaliation}%
                  </CircularProgressLabel>
                </CircularProgress>
              </Box>
              <Text color="white" fontSize="sm" mt={2}>
                Avaliação dos usuários
              </Text>
            </HStack>

            {overview && (
              <>
                <Heading as="h5" size="md" color="white" mt={8}>
                  Sinopse
                </Heading>{' '}
                <Text
                  noOfLines={5}
                  fontSize="1rem"
                  lineHeight={6}
                  color="#dddddd"
                  mt={2}
                  data-testid="banner-movie-overview"
                >
                  {overview}
                </Text>
              </>
            )}
            <HStack gap="60px" mt={6} data-testid="banner-movie-directors">
              {credits.map((credit) => (
                <VStack key={credit.name} alignItems="flex-start" gap={0}>
                  <Text color="white" fontWeight={700} fontSize="1rem">
                    {credit.name}
                  </Text>
                  <Text color="white" fontWeight={400} fontSize="0.875rem">
                    {credit.type}
                  </Text>
                </VStack>
              ))}
            </HStack>
            <HStack gap="60px" mt={6} data-testid="banner-movie-screenplay">
              {screenPlay.map((credit) => (
                <VStack key={credit.name} alignItems="flex-start" gap={0}>
                  <Text color="white" fontWeight={700} fontSize="1rem">
                    {credit.name}
                  </Text>
                  <Text color="white" fontWeight={400} fontSize="0.875rem">
                    {credit.type}
                  </Text>
                </VStack>
              ))}
            </HStack>
          </VStack>
        </HStack>
      </Container>
    </Box>
  );
};

export default BannerMovie;
