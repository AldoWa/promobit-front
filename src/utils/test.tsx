import { RenderOptions, render } from '@testing-library/react'

import { ChakraProvider } from '@chakra-ui/react'

interface AllTheProvidersProps {
  children: React.ReactNode
}

const AllTheProviders = ({children}: AllTheProvidersProps) => {
  return (
    <ChakraProvider>
      {children}
    </ChakraProvider>
  )
}

const customRender = (ui:React.ReactElement, options?: RenderOptions) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// eslint-disable-next-line react-refresh/only-export-components
export * from '@testing-library/react'

// override render method
export {customRender as render}