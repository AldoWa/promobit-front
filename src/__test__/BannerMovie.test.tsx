import { render, screen } from '../utils/test';
import BannerMovie from '../components/BannerMovie';

jest.mock('../components/Filter')

describe('Banner', () => {
  it("Should render the Banner movie correctly", async () => {
    render(<BannerMovie 
      title="Teste Title"
      imageUrl="https://image.tmdb.org/t/p/original/9yBVqNruk6Ykrwc32qrK2TIE5xw.jpg"
      credits={[{ type: 'Director', name: 'Ator 1' },{ type: 'Director', name: 'Ator 2' }]}
      overview="Teste Overview"
      screenPlay={[{ type: 'Director', name: 'Ator 1' }]}
      subtitle="Teste | Subtitle"
      avaliation={10}
    />)
    const container = screen.getByTestId('banner-movie-container')
    expect(container).toBeInTheDocument()

    const title = screen.getByTestId('banner-movie-title')
    expect(title).toHaveTextContent('Teste Title')

    const subtitle = screen.getByTestId('banner-movie-subtitle')
    expect(subtitle).toHaveTextContent('Teste | Subtitle')

    const overview = screen.getByTestId('banner-movie-overview')
    expect(overview).toHaveTextContent('Teste Overview')

    const credits = screen.getByTestId('banner-movie-directors').childNodes
    expect(credits.length).toBe(2)

    const screenplay = screen.getByTestId('banner-movie-screenplay').childNodes
    expect(screenplay.length).toBe(1)

    const avaliation = screen.getByTestId('banner-movie-avaliation')
    expect(avaliation).toHaveTextContent('10%')
  })

  it("Should match snapshot", async () => {
    const { container } = render(
    <BannerMovie 
      title="Teste Title"
      imageUrl="https://image.tmdb.org/t/p/original/9yBVqNruk6Ykrwc32qrK2TIE5xw.jpg"
      credits={[{ type: 'Director', name: 'Ator 1' },{ type: 'Director', name: 'Ator 2' }]}
      overview="Teste Overview"
      screenPlay={[{ type: 'Director', name: 'Ator 1' }]}
      subtitle="Teste | Subtitle"
      avaliation={10}
    />)
    expect(container).toMatchSnapshot();
  })
})