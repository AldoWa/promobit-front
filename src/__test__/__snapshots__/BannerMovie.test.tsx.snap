// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Should match snapshot 1`] = `
<div>
  <div
    class="css-pnnn0m"
    data-testid="banner-movie-box"
  >
    <div
      class="chakra-container css-1kw8fss"
      data-testid="banner-movie-container"
    >
      <div
        class="chakra-stack css-jlor57"
      >
        <div
          class="chakra-skeleton css-m149th"
        />
        <div
          class="chakra-stack css-mw8ufx"
        >
          <h3
            class="chakra-heading css-y5314g"
            data-testid="banner-movie-title"
          >
            Teste Title
          </h3>
          <p
            class="chakra-text css-vtk0uk"
            data-testid="banner-movie-subtitle"
          >
            Teste | Subtitle
          </p>
          <div
            class="chakra-stack css-ut3x8m"
          >
            <div
              class="css-hmral"
            >
              <div
                aria-valuemax="100"
                aria-valuemin="0"
                aria-valuenow="10"
                class="chakra-progress css-1kj616h"
                role="progressbar"
              >
                <svg
                  class="css-1k0co84"
                  viewBox="0 0 100 100"
                >
                  <circle
                    class="chakra-progress__track css-nj4szp"
                    cx="50"
                    cy="50"
                    r="42"
                    stroke-width="10px"
                  />
                  <circle
                    class="chakra-progress__indicator css-xowg6v"
                    cx="50"
                    cy="50"
                    r="42"
                    stroke-dasharray="26.400000000000002 237.6"
                    stroke-dashoffset="66"
                    stroke-width="10px"
                  />
                </svg>
                <div
                  class="css-158rzxq"
                  data-testid="banner-movie-avaliation"
                >
                  10
                  %
                </div>
              </div>
            </div>
            <p
              class="chakra-text css-1u8wf1z"
            >
              Avaliação dos usuários
            </p>
          </div>
          <h5
            class="chakra-heading css-1k02bmm"
          >
            Sinopse
          </h5>
           
          <p
            class="chakra-text css-1r0a2rl"
            data-testid="banner-movie-overview"
          >
            Teste Overview
          </p>
          <div
            class="chakra-stack css-mqdrn6"
            data-testid="banner-movie-directors"
          >
            <div
              class="chakra-stack css-enj3l8"
            >
              <p
                class="chakra-text css-1xzcys3"
              >
                Ator 1
              </p>
              <p
                class="chakra-text css-14om7b3"
              >
                Director
              </p>
            </div>
            <div
              class="chakra-stack css-enj3l8"
            >
              <p
                class="chakra-text css-1xzcys3"
              >
                Ator 2
              </p>
              <p
                class="chakra-text css-14om7b3"
              >
                Director
              </p>
            </div>
          </div>
          <div
            class="chakra-stack css-mqdrn6"
            data-testid="banner-movie-screenplay"
          >
            <div
              class="chakra-stack css-enj3l8"
            >
              <p
                class="chakra-text css-1xzcys3"
              >
                Ator 1
              </p>
              <p
                class="chakra-text css-14om7b3"
              >
                Director
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
